Submitted By:            Fernando de Oliveira <famobr at yahoo dot com dot br>
Date:                    2015-12-25
Initial Package Version: xorg-server-1.18.0 (for tigervnc-1.6.0)
Upstream Status:         Not Submitted
Origin:                  ArchLinux
Description:             Allow using Xorg-server-1.18.0
================================================================================


================================================================================

diff -wbBur tigervnc-1.5.0/unix/xserver/hw/vnc/Input.c tigervnc-1.5.0.q/unix/xserver/hw/vnc/Input.c
--- tigervnc-1.5.0/unix/xserver/hw/vnc/Input.c	2015-07-11 16:00:36.000000000 +0300
+++ tigervnc-1.5.0.q/unix/xserver/hw/vnc/Input.c	2015-11-25 19:04:24.278747038 +0300
@@ -300,6 +300,8 @@
 #if XORG < 111
 	n = GetKeyboardEvents(eventq, dev, action, kc);
 	enqueueEvents(dev, n);
+#elif XORG > 117
+	QueueKeyboardEvents(dev, action, kc);
 #else
 	QueueKeyboardEvents(dev, action, kc, NULL);
 #endif
diff -wbBur tigervnc-1.5.0/unix/xserver/hw/vnc/xorg-version.h tigervnc-1.5.0.q/unix/xserver/hw/vnc/xorg-version.h
--- tigervnc-1.5.0/unix/xserver/hw/vnc/xorg-version.h	2015-07-11 16:00:36.000000000 +0300
+++ tigervnc-1.5.0.q/unix/xserver/hw/vnc/xorg-version.h	2015-11-25 19:02:47.688751421 +0300
@@ -48,8 +48,10 @@
 #define XORG 116
 #elif XORG_VERSION_CURRENT < ((1 * 10000000) + (17 * 100000) + (99 * 1000))
 #define XORG 117
+#elif XORG_VERSION_CURRENT < ((1 * 10000000) + (18 * 100000) + (99 * 1000))
+#define XORG 118
 #else
-#error "X.Org newer than 1.17 is not supported"
+#error "X.Org newer than 1.18 is not supported"
 #endif
 
 #endif
